crates/torii/grpc/proto/world.proto

```diff
@@ -24,3 +34,71 @@ message MetaReply {
     // The hex-encoded class hash of the executor.
     string executor_class_hash = 4;
 }
+
+// A request to retrieve a component value of an entity.
+message GetEntityRequest {
+    // The component name.
+    string component = 1;
+    // The entity keys.
+    EntityKeys keys = 2;
+}
+
+// The entity response contains the component values for the requested entities.
+message GetEntityResponse {
+    repeated string values = 1;
+}
+
+// A request to retrieve all entities of a component.
+message GetEntitiesRequest {
+    // The component name.
+    string component = 1;
+}
+
+// The entities response contains the keys of all entities for the requested component.
+message GetEntitiesResponse {
+    // The list of entities keys.
+    repeated EntityKeys keys = 1;
+}
+
+// The entity keys which is a list of hex-encoded field elements.
+message EntityKeys {
+    repeated string values = 1;
+}
+
+// The entity values which is a list of hex-encoded field elements.
+message EntityValues {
+    repeated string values = 2;
```

Comment:
When a state diff is generated that only modifies a subset of a components members, the diff will only include those values. For example, if we have `struct Postion { x, y }` and only `y` is updated, the diff will have a single slot as being modified. So we will need to know the values position offset as well, so we can do a partial update on the client.

Reply:
yeah for this i was just trying to imagine that maybe the state diff can include all the values even though only 1 slot is changed to make applying the diffs easier. but yeah maybe doesn't really make sense to include unchanged slots in the diffs

Reply:
yeah since it will require additional reads to do that too. one question is to what extent do we want to preprocess the state diff in torii. maybe we should pass the raw state diff filtered by the entities the client is subscribed to and handle the parsing into component schema in torii client

Reply:
We don't need to access the events or sql db at all. We should be able to use the state diff on the block directly:

When a client subscribes to a set of entity components, we compute the storage keys for all of those values and store them in the filter. When a new block is produced and handled by the indexer, we take the blocks state diff and filter it using the client filter, then forward the filtered state diff to the torii client to handle